content/en/guides/features/live-preview.md

```diff
@@ -64,17 +64,30 @@ enablePreview should be tested locally with yarn start and not yarn
 dev
 </base-alert>
 
-For pages that are not yet generated, SPA fallback will still call the API before showing the 404 page as these pages exist on the API but are not re-generated yet
+### Previewing pages that are not yet generated
 
-For pages that are not generated and don't exist on the API, for example draft content in your CMS that is not publicly available, you will need to use the validate hook:
+For pages that are not yet generated, SPA fallback will still call the API before showing the 404 page as these pages exist on the API but are not generated yet.
+
+If you have set a validate hook, you will probably need to modify it so that it doesn't redirect to the 404 page in preview mode.
 
 ```js
-async validate({ app, params, $preview }) {
-  if ($preview) {
+validate({ params, query }) {
+  if (query.preview) {
     return true
 }
 ```
 
+### Passing data to enablePreview
+
+You can pass data to the `enablePreview` function. That data will then be available on the `$preview` context helper and on `this.$preview`.
+This can be useful if you need to fetch data differently in preview mode (e.g. making authenticated requests to fetch draft content).
+
+```js
+asyncData ({ $preview }) {
+  if ($preview) return { page: $preview }
```

Comment:
I don't think this example returning `{ page: $preview }` is correct, instead you can fake an api call and give a token from `$preview` instead.

Reply:
You mean you would call `enablePreview({token: 'xxxxxx'})` from the plugin and then access that token inside asyncData? In my example, I assume that the API call with the token is already made inside the plugin and then the data is passed to `enablePreview(data)`.

Reply:
I see, in that case I think we should add more code example for this section to show the plugin and maybe some comments.

Also, I think if you use $axios, you can directly set the token in the preview plugin instead of using `$preview`

Reply:
I now wonder if making auth request to preview draft content in headless CMS is a good use case for this feature.
What use cases did you have in mind when adding the `$preview` context helper?

Reply:
Actually I don't have such at the moment 😅

Reply:
😅 In that case it's maybe best to only mention that it's possible to pass data, without giving any example.

Reply:
Yes, can you update it :)